packages/zoe/src/contracts/loan/liquidate.js

```diff
@@ -0,0 +1,59 @@
+// @ts-check
+import '../../../exported';
+
+import { E } from '@agoric/eventual-send';
+
+import { depositToSeat, withdrawFromSeat } from '../../contractSupport';
+
+/**
+ * This function is triggered by the priceOracle when the value of the
+ * collateral is below the mmr percentage. The function performs the
+ * liquidation and then shuts down the contract. Note that if a
+ * liquidation occurs, the borrower gets nothing and they can take no
+ * further action.
+ *
+ * @type {Liquidate}
+ */
+export const liquidate = async (zcf, config, expectedValue) => {
+  const { collateralSeat, autoswap, lenderSeat } = config;
+
+  // For simplicity, we will sell all collateral.
+  const zoeService = zcf.getZoeService();
+
+  const allCollateral = collateralSeat.getAmountAllocated('Collateral');
+
+  // TODO: add some buffer in case the price changes?
+  const { Collateral: collateralPayment } = await withdrawFromSeat(
+    zcf,
+    collateralSeat,
+    {
+      Collateral: allCollateral,
+    },
+  );
+
+  const proposal = harden({
+    want: { Out: expectedValue },
+    give: { In: allCollateral },
+  });
+
+  const payments = harden({ In: collateralPayment });
+
+  const swapInvitation = E(autoswap).makeSwapInInvitation();
+  const autoswapSeat = E(zoeService).offer(swapInvitation, proposal, payments);
```

Comment:
This is a different kind of seat than `collateralSeat` and `lenderSeat` (which explains why it doesn't need a `.exit()` call at the end.) Can we name it differently so it doesn't look so much like the same kind of thing?

Reply:
Renamed to `autoswapUserSeat`